CMakeLists.txt

```diff
@@ -6,28 +6,58 @@ endif()
 
 cmake_minimum_required(VERSION 3.0)
 project(Luau LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 17)
 
 option(LUAU_BUILD_CLI "Build CLI" ON)
+option(LUAU_BUILD_WEB_REPL "Build Web CLI" ON)
 option(LUAU_BUILD_TESTS "Build tests" ON)
 
 add_library(Luau.Ast STATIC)
 add_library(Luau.Compiler STATIC)
 add_library(Luau.Analysis STATIC)
 add_library(Luau.VM STATIC)
 
-if(LUAU_BUILD_CLI)
-    add_executable(Luau.Repl.CLI)
-    add_executable(Luau.Analyze.CLI)
+if(LUAU_BUILD_WEB_REPL AND EMSCRIPTEN)
```

Comment:
I'm hoping that this entire setup can be much simpler if we go with `__EMSCRIPTEN__` suggestion from above. In the same vein, here we could add the necessary target options to Luau.Repl.CLI if `EMSCRIPTEN` is set. I don't think we need to disable targets manually, as that's easy to do from CMake command line.